code/LeftAndMain.php

```diff
@@ -864,9 +864,17 @@ public function redirect($url, $code = 302)
         if ($this->getRequest()->isAjax()) {
             $response = $this->getResponse();
             $response->addHeader('X-ControllerURL', $url);
+
+            /** Since in some cases we need to pass additional headers,
+             * but we lose these headers (X-Pjax and X-Status) when we set new response
+             * we should explicitly add them again from received request.
```

Comment:
> when we set new response

You said in https://github.com/silverstripe/silverstripe-admin/pull/1615#discussion_r1379395076 that it's to do with a redirect, but here you're saying it's when we set a new response.

If it's because we're using `$this->setResponse($someNewResponse)`, then we have full control over the headers in the new response and should apply them correctly.

If it's because of a redirection, please change the wording in the comment to mention a redirection, and not to mention setting a new response.

Reply:
I'm not sure I understand what you mean. But I think it has the right meaning. We send a redirection request with a link to where the user needs to be redirected. In doing so, we create a completely new response. But we also need to save some headers from the request.

Reply:
Removed this block